File: cognitive-exercises-backend/cognitive-exercises-data/src/main/java/com/zelazobeton/cognitiveexercieses/model/ScoreboardPageDto.java
package com.zelazobeton.cognitiveexercieses.model;

import java.util.List;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class ScoreboardPageDto {
    List<UserScoreDto> userScores;
    int pageNumber;
    int pagesTotal;

    public ScoreboardPageDto(List<UserScoreDto> userScores, int pageNumber, int pagesTotal) {
        this.userScores = userScores;
        this.pageNumber = pageNumber;
        this.pagesTotal = pagesTotal;
    }
}

File: cognitive-exercises-frontend/src/app/shared/model/scoreboard-page-dto.ts
import {PortfolioDto} from './portfolio-dto';

export class ScoreboardPageDto {
  public userScores: UserScoreDto[];
  public pageNumber: number;
  public pagesTotal: number;

  constructor(userScores: UserScoreDto[], pageNumber: number, pagesTotal: number) {
    this.userScores = userScores;
    this.pageNumber = pageNumber;
    this.pagesTotal = pagesTotal;
  }
}

export class UserScoreDto {
  public place: number;
  public username: string;
  public portfolio: PortfolioDto;

  constructor(place: number, username: string, portfolio: PortfolioDto) {
    this.place = place;
    this.username = username;
    this.portfolio = portfolio;
  }
}